enterprise/functions/src/main/java/io/crate/window/LagFunction.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * This file is part of a module with proprietary Enterprise Features.
+ *
+ * Licensed to Crate.io Inc. ("Crate.io") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Unauthorized copying of this file, via any medium is strictly prohibited.
+ *
+ * To use this file, Crate.io must have given you permission to enable and
+ * use such Enterprise Features and you must have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  If you enable or use the Enterprise
+ * Features, you represent and warrant that you have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  Your use of the Enterprise Features
+ * if governed by the terms and conditions of your Enterprise or Subscription
+ * Agreement with Crate.io.
+ */
+
+package io.crate.window;
+
+import io.crate.data.Input;
+import io.crate.data.Row;
+import io.crate.data.RowN;
+import io.crate.execution.engine.collect.CollectExpression;
+import io.crate.execution.engine.window.WindowFrameState;
+import io.crate.execution.engine.window.WindowFunction;
+import io.crate.metadata.BaseFunctionResolver;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionImplementation;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.functions.params.FuncParams;
+import io.crate.metadata.functions.params.Param;
+import io.crate.module.EnterpriseFunctionsModule;
+import io.crate.types.DataType;
+
+import java.util.List;
+
+public class LagFunction implements WindowFunction {
+
+    private static final String NAME = "lag";
+
+    private final FunctionInfo info;
+
+    private LagFunction(FunctionInfo info) {
+        this.info = info;
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return info;
+    }
+
+    @Override
+    public Object execute(int idxInPartition,
+                          WindowFrameState currentFrame,
+                          List<? extends CollectExpression<Row, ?>> expressions,
+                          Input... args) {
+
+        final int offset;
+        if (args.length > 1) {
+            offset = ((Number) args[1].value()).intValue();
+        } else {
+            offset = 1;
+        }
+
+        if (offset <= idxInPartition) {
+            int lagIdx = idxInPartition - offset;
+            var lagRowCells = currentFrame.getRowInPartitionAtIndexOrNull(lagIdx);
+            if (lagRowCells == null) {
+                return null;
+            }
+
+            Row lagRow = new RowN(lagRowCells);
+            for (CollectExpression<Row, ?> expression : expressions) {
+                expression.setNextRow(lagRow);
+            }
+
+            return args[0].value();
+        } else {
+            if (args.length == 3) {
```

Comment:
Given that the behavior / args description is a bit disconnected from the implementation (only in the docs) - maybe we could add some short descriptions somewhere. E.g. repeat the synposis on class level or something. So it's easier to understand that the third argument is optional and the default value.